Docs/Book/GettingStartedInC++.md

```diff
@@ -0,0 +1,1420 @@
+change within the next year.
+
+C++ programs generally have quite a lot of extra verbiage in them
+compared to, for example, python scripts, that is similar or identical
+in all programs.  The would produce extraneous clutter in the example
+code in this document.  Because of this, only the minimum code to
+exemplify the point being made is shown in the text. Full programs for
+all the examples (generally with multiple examples in the same
+program) are given in the `$RDBASE/Docs/Book/C++Examples` directory in
+the distribution. The particular program will be mentioned in the text.
+
+The functions of the RDKit system are declared in a large number of
+different header files spread across several directories in the
+system, and defined across a number of different libraries.  The right
+headers will need to be included in the source code, and libraries
+linked to during linking.  Whilst it's possible to include all headers
+and libraries in all executables this will result in bloated
+executables which will take up a lot more disk and memory, especially
+if you are doing static linking.  When linking to the static (.a)
+libraries rather than the shared-object (.so) ones, the order the
+libraries appear in linking list can be important.  See the
+CMakeLists.txt file in C++Examples directory for a good order.  In
+this case, the same library list is used for all examples, so some
+will be unnecessary for some of the programs. The first 3 programs
+don't need the Depictor and SubstructMatch libraries, for instance,
+although on my Ubuntu 16.04 system, the RDGeometryLib appears to need
+to be included twice. Working out which libraries need to be linked to
+and in what order can involve a tedious amount of trial and error.
+
+## Should You Use C++ or Python?
+
+There is no doubt that it is much easier get started with
+Python.  If you follow the installation instructions, you will be able
+to start programming and using scripts straightaway.  If all you are
+going to do is use scripts to do relatively simple things,
+essentially stitching RDKit function calls together, there should be
+little or no speed issues with using the Python interpreted language,
+as all the RDKit functions are compiled C++ and well optimised.
+However, if you are going to do more complicated things, using a lot
+of your own programming logic and only using the RDKit for peripheral
+things like I/O, SMARTS matching, preparing 2D images and the like,
+then it is likely that you will have good performance gains if you
+write in C++ and compile to a native executable.
+
+Another consideration is the completeness of the API.  A lot of the
+higher level functionality in RDKit is developed in Python, and
+back-porting to C++ occurs on a demand-driven basis.  There are
+therefore examples of quite useful functionality, such as drawing
+multiple 2D molecule diagrams to a grid, or computing the RMS
+differences between conformers that are not available in C++.  Of
```

Comment:
RMS difference in C++: This has nothing to do with the PR, this functionality will be available in C++, once I get my rewrite of the MolAlign code added. This rewrite moves all the MolAlign functionality from Python into C++.